src/libraries/Common/src/System/Net/Mail/MailBnfHelper.cs

```diff
@@ -267,7 +267,7 @@ internal static string ReadToken(string data, ref int offset, StringBuilder? bui
                 }
             }
 
-            if (start == offset)
+            if (start == offset && offset < data.Length)
```

Comment:
If `offset < data.Length` then the previous `for` loop must have ended in `else if (!Ttext[data[offset]])`.
Before this change, we would substring the data up until the first non-token character. And now it seems we're throwing?
This doesn't seem right.